File: python/dave_game.py
import random

random_number = random.randint(0, 10)

while True:
    guess = int(raw_input("Guess a number (0 - 10):"))

    if guess > random_number:
      print "Too high! Try again.\n"
    elif guess < random_number:
      print "Too low! Try again.\n"
    elif guess == random_number:
      print "You guessed it! Goodbye.\n"
      break
    else:
      print "That's not a valid number.\n"

File: javascript/dave_game.js
var random_number = Math.floor(Math.random() * 11);

while (true) {
  var guess = prompt("Guess a number (0 - 10):");

  if (guess > random_number) {
    alert("Too high! Guess again.");
  } else if (guess < random_number) {
    alert("Too low! Guess again.");
  } else if (guess == random_number) {
    alert("You guessed it! Goodbye.");
    break;
  } else {
    alert("That is not a valid number.");
  }
}